test/e2e/workflow/deployment_test.go

```diff
@@ -181,16 +181,26 @@ var _ = Describe("NetworkAddonsConfig", func() {
 })
 
 func testConfigCreate(configSpec opv1alpha1.NetworkAddonsConfigSpec, components []Component) {
-	CreateConfig(configSpec)
-	checkConfigChange(components)
+	checkConfigChange(components, func() {
+		CreateConfig(configSpec)
+	})
 }
 
 func testConfigUpdate(configSpec opv1alpha1.NetworkAddonsConfigSpec, components []Component) {
-	UpdateConfig(configSpec)
-	checkConfigChange(components)
+	checkConfigChange(components, func() {
+		UpdateConfig(configSpec)
+	})
 }
 
-func checkConfigChange(components []Component) {
+func checkConfigChange(components []Component, while func()) {
```

Comment:
I would add a TODO comment to change this once controler-runtime support monitor functionality

Reply:
Good call, opened an RFE: https://github.com/operator-framework/operator-sdk/issues/2655